### DISCUSSION TITLE
Plotting Altitude as Height Coordinate for Aircraft Radar Data

### DISCUSSION THREAD

**Discussion by evannewman** [ARM-Development/RadTraQ#60] [Q&A]
Hello,

I am wondering how one might be able to plot the height variable in respect to altitude and not range? For example, the sample dataset and CFAD code has a variable stored as "alt". When I pass in "alt" as the height variable for the calc_cfad function, the following error occurs:

"coordinate alt has dimensions ('range',), but these are not a subset of the DataArray dimensions ('alt', 'x')"

How would one be able to get fix this problem?

Evan

**Comment by AdamTheisen** [comment]
Hello @evannewman,

I can take a look at this in more detail.  Can you provide the file you're working with?

**Reply by evannewman** [reply]
Hi Adam, thank you for the quick response. The sample file 'sgpkazrgeC1.a1.20190529.000002.cdf' from the open_radar_data.DATASETS module (the same data that is used in the example CFAD plotting code on the RadTraq documentation site) could be used to see the issue, and then adding the "alt" variable as the height variable in the calc_cfad call. The altitude variable is a function of range only. I have radar aircraft data in which I am attempting to plot cfads for and, being aircraft data, I don't want to plot on range but rather altitude above MSL.

**Comment by AdamTheisen** [comment]
Thanks @evannewman, that is helpful!

With the example file, this might be an issue with how we subset the data using xarray.  The 'alt' variable in the ARM files is typically a scalar with a single value of the altitude of the instrument.  If you look at the 'alt' variable it's just a value of 316 repeated.  I did find a bug that I will get a fix in for but utilizing 'alt' as the height variable with this example file will produce a blank plot but it should work for other datasets where altitude is the altitude of the data.  Hopefully I'm not just rambling and that makes sense!

**Comment by AdamTheisen** [comment]
I should note that if you would like to add some example aircraft data to the open-radar-data repo, it would be very helpful and we can craft up some examples around it as well.

https://github.com/openradar/open-radar-data

**Reply by evannewman** [reply]
Hi again Adam! I wanted to bring up this issue again with the cfads plotting on altitude vs range. I still was unable to plot with altitude as the height variable both on the sample dataset and my own aircraft radar data. Provided is a screenshot of the same error I get for both datasets when adding the "height variable = 'altitude'" keyword to the cfad calculation function. Both datasets hold altitude with the dimension of "range", thus the issue isn't that the sample ARM data has essentially null altitude data, but rather the Xarray DataArray format is unable to parse the extra call in the cfad calculation function.

![cfads_error](https://github.com/ARM-Development/RadTraQ/assets/122176400/a9b9ce40-54bc-44f9-b2ba-16a6bb394a9d)
![cfads_error2](https://github.com/ARM-Development/RadTraQ/assets/122176400/41fd0c55-75bf-4064-b00b-a0312936fc4e)

**Reply by AdamTheisen** [reply]
@evannewman are you able to point me to a sample of your data?  Also, did you clone the GitHub repo or install through pip or conda?

**Reply by evannewman** [reply]
Adam, attached is a sample file I am using. The reflectivity data, along with all other important variables, is stored in the netcdf group name "sweep_up-down". And I installed radtraq through pip.

Using the API source code yesterday, I was able to convert the calc_cfad and plot_cfad functions to using altitude exclusively (as that is what's most important for my research). However, like I said it was more so a brute force method rather than tackling the actual problem I was experiencing with the Radtraq built in functions.

[Uploading cfrad2.20170218_215817_to_20170218_222206_WCR_L2_SNOWIE17.CPP-H1V2H2.up-down.zip…]()

**Reply by AdamTheisen** [reply]
Thanks @evannewman !  I'll test this out but I did just merge in an update but we did not do a release yet.  After I'm done testing I'll do a micro release.

**Reply by AdamTheisen** [reply]
@evannewman just as a note, I have not been able to download this file.  It keeps bringing me back to this discussion page instead of downloading.  I'm going to go forward with a release and see if the latest bug fix I put in will work for you.

**Reply by AdamTheisen** [reply]
@evannewman just checking in, have you had a chance to try the latest version?  

I should note that Py-ART has a [CFAD algorithm](https://arm-doe.github.io/pyart/examples/retrieve/plot_cfad.html#sphx-glr-examples-retrieve-plot-cfad-py) as well and we are going to be discussing if it makes sense to [roll up the RadTraQ functionality into Py-ART](https://github.com/ARM-DOE/pyart/issues/1522).

**Reply by evannewman** [reply]
Hi Adam, sorry for the late response. I have not yet had the chance to try the latest version of RadTraq with the updates. Once I used the source code to get what I needed to change to plotting on altitude, I moved on. But let me add this to my list and I will get back to you and let you know if the error/bug still arises! Thank you again for your help!
